ompi/errhandler/errcode.c

```diff
@@ -34,84 +36,84 @@
 #include "ompi/constants.h"
 
 /* Table holding all error codes */
-opal_pointer_array_t ompi_mpi_errcodes;
+opal_pointer_array_t ompi_mpi_errcodes = {{0}};
 int ompi_mpi_errcode_lastused=0;
 int ompi_mpi_errcode_lastpredefined=0;
 
-ompi_mpi_errcode_t ompi_success;
-ompi_mpi_errcode_t ompi_err_buffer;
-ompi_mpi_errcode_t ompi_err_count;
-ompi_mpi_errcode_t ompi_err_type;
-ompi_mpi_errcode_t ompi_err_tag;
-ompi_mpi_errcode_t ompi_err_comm;
-ompi_mpi_errcode_t ompi_err_rank;
-ompi_mpi_errcode_t ompi_err_request;
-ompi_mpi_errcode_t ompi_err_root;
-ompi_mpi_errcode_t ompi_err_group;
-ompi_mpi_errcode_t ompi_err_op;
-ompi_mpi_errcode_t ompi_err_topology;
-ompi_mpi_errcode_t ompi_err_dims;
-ompi_mpi_errcode_t ompi_err_arg;
-ompi_mpi_errcode_t ompi_err_unknown;
-ompi_mpi_errcode_t ompi_err_truncate;
-ompi_mpi_errcode_t ompi_err_other;
-ompi_mpi_errcode_t ompi_err_intern;
-ompi_mpi_errcode_t ompi_err_in_status;
-ompi_mpi_errcode_t ompi_err_pending;
+ompi_mpi_errcode_t ompi_success = {{0}};
+ompi_mpi_errcode_t ompi_err_buffer = {{0}};
```

Comment:
I think all these error codes can be `static` instead of initializing them.  It looks like the only way they are visible externally to this file is by having pointers to them stuffed into the `ompi_mpi_errcodes` array.

Obliquely related note: it's lame that we have two separate variables named `ompi_err_buffer` (here and in `ompi/errhandler/errcode-internal.c`, the latter is already `static`).  I know that Gilles didn't do that, I just noticed it while checking for `static` opportunities.